File: lesson-1-linear-search/zig/linear_search.zig
const std = @import("std");

pub fn linearSearch(arr: []i32, target: i32) !usize {
    var i: usize = 0;
    while (i < arr.len) : (i += 1) {
        if (arr[i] == target) {
            return i;
        }
    }
    return error.NoElementFound;
}

pub fn main() !void {
    const arr: []i32 = [3, 45, 1, 2, 99];
    const target: i32 = 1;
    var result: usize = try linearSearch(arr, target);
    if (result != error.NoElementFound) {
        std.debug.print("Element found at index: {}\n", .{result});
    } else {
        std.debug.print("Element not found in the array\n");
    }
}

File: lesson-1-linear-search/typescript/linear_search.ts
function linearSearch(arr: number[], target: number): number {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === target) {
            return i;
         } 
     }
    return -1; 
 }

//  Test 1
const arr: number[] = [4, 2, 7, 1, 3];
const target: number = 7;

const result: number = linearSearch(arr, target);
if (result !== -1) {
    console.log(`Element founh at index ${result}`);
} else {
    console.log("element does not exist in the array");
 }
